rpm2extents.c

```diff
@@ -239,181 +277,219 @@ static rpmRC process_package(FD_t fdi, FD_t digestori, FD_t validationi)
     uint32_t offset_ix = 0;
     size_t len;
     int next = 0;
+    struct rpmlead_s l;
+    rpmfiles files = NULL;
+    rpmfi fi = NULL;
+    char *msg = NULL;
 
     fdo = fdDup(STDOUT_FILENO);
 
-    if (rpmReadPackageRaw(fdi, &sigh, &h)) {
-	rpmlog(RPMLOG_ERR, _("Error reading package\n"));
-	exit(EXIT_FAILURE);
-    }
+    rc = rpmLeadReadAndReturn(fdi, &msg, &l);
+    if (rc != RPMRC_OK)
+	goto exit;
 
-    if (rpmLeadWrite(fdo, h))
-    {
-	rpmlog(RPMLOG_ERR, _("Unable to write package lead: %s\n"),
-		Fstrerror(fdo));
-	exit(EXIT_FAILURE);
-    }
+    /* Skip conversion if package is in deny list */
+    if (isInDenyList(l.name)) {
+	if (rpmLeadWrite(fdo, l)) {
+	    fprintf(stderr, _("Unable to write package lead: %s\n"),
+		    Fstrerror(fdo));
+	    rc = RPMRC_FAIL;
+	    goto exit;
+	}
 
-    if (rpmWriteSignature(fdo, sigh)) {
-	rpmlog(RPMLOG_ERR, _("Unable to write signature: %s\n"), Fstrerror(fdo));
-	exit(EXIT_FAILURE);
-    }
+	ssize_t fdilength = ufdCopy(fdi, fdo);
+	if (fdilength == -1) {
+	    fprintf(stderr, _("process_package cat failed\n"));
+	    rc = RPMRC_FAIL;
+	    goto exit;
+	}
 
-    if (headerWrite(fdo, h, HEADER_MAGIC_YES)) {
-	rpmlog(RPMLOG_ERR, _("Unable to write headers: %s\n"), Fstrerror(fdo));
-	exit(EXIT_FAILURE);
-    }
+	goto exit;
+    } else {
+	if (rpmReadPackageRaw(fdi, &sigh, &h)) {
```

Comment:
that's a lot of green code below. I expect there is no change other than being wrapped within the else statement and some formatting changes.

Reply:
yes, it's only indentation change after that.